src/main/java/com/redhat/devtools/intellij/common/actions/component/RunTaskAction.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.common.actions.component;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.redhat.devtools.intellij.common.actions.TknAction;
+import com.redhat.devtools.intellij.common.utils.ExecHelper;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TasksNode;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+
+public class RunTaskAction extends TknAction {
+    public RunTaskAction() { super(TaskNode.class); }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, String tkncli) throws IOException {
+        ExecHelper.execute(tkncli, "task", "start", selected.toString(), "-n", ((TaskNode)selected).getParent().getParent().toString());
```

Comment:
Every run of the tkn cli should be done in the TknCli class and not spread all over the code. Also as a new task run will be created, you must class refresh on the selected object so that the list will be refreshed. And the command string should be hidden and Tkn object should be given to any action

Reply:
Reloading the selected object doesn't solve the issue of having an updated list. The last launched taskrun doesn't appear even if we reload the node (creating a pod and run the task requires some time) and, in any case, the user will see the task in a running state and the tree won't refresh when the task succeed or fail. So the resulting list will always be not updated